MediaBrowser.Controller/Entities/Folder.cs

```diff
@@ -1333,8 +1333,8 @@ private int AddChildren(User user, bool includeLinkedChildren, Dictionary<Guid,
         private int AddChildrenFromCollection(IEnumerable<BaseItem> children, User user, bool includeLinkedChildren, Dictionary<Guid, BaseItem> result, bool recursive, InternalItemsQuery query, HashSet<Folder> visitedFolders)
         {
             query ??= new InternalItemsQuery();
-            var limit = query.Limit;
-            query.Limit = 100; // this is a bit of a dirty hack thats in favor of specifically the webUI as it does not show more then +99 elements in its badges so there is no point in reading more then that.
+            var limit = query.Limit.HasValue && query.Limit > 0 ? query.Limit : int.MaxValue;
```

Comment:
I think this can be simplified to 

query.Limit > 0 ? query.Limit : int.MaxValue